rust/hyperlane-core/src/traits/cursor.rs

```diff
@@ -2,22 +2,29 @@ use std::{fmt, ops::RangeInclusive, time::Duration};
 
 use async_trait::async_trait;
 use auto_impl::auto_impl;
+use eyre::Result;
 
-use crate::{ChainResult, LogMeta};
+use crate::LogMeta;
 
 /// A cursor governs event indexing for a contract.
 #[async_trait]
 #[auto_impl(Box)]
 pub trait ContractSyncCursor<T>: Send + Sync + 'static {
     /// The next block range that should be queried.
-    async fn next_action(&mut self) -> ChainResult<(CursorAction, Duration)>;
+    /// This method should be tolerant to being called multiple times in a row
+    /// without any updates in between.
+    async fn next_action(&mut self) -> Result<(CursorAction, Duration)>;
 
-    /// The latest block that has been queried
+    /// The latest block that has been queried, used as a proxy for health.
+    /// TODO: consider a better way to assess health
     fn latest_block(&self) -> u32;
```

Comment:
Should it also include a timestamp along with the block number? =

Reply:
added to https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3265